In [ ]:
class Estadisticos:
  def __init__(self,x1):
    self.x1=x1
  def media(self):
    return sum(self.x1)/len(self.x1)
  def mediana(self):
    self.x1.sort()
    num= len(self.x1)
    if num %2==0:
      print((self.x1[int(num/2)]+self.x1[int(num/2)-1])/2)
    else:
      print(self.x1[int(num/2-0.5)])
  def varianza(self):
    media_x=self.media()
    vec=[(i-media_x)**2 for i in self.x1]
    return sum(vec)/(len(vec)-1)
  def desviacion_estandar(self):
    return self.varianza()**(1/2)
  def coeficiente_variacion(self):
    return self.desviacion_estandar()/self.media()

In [ ]:
!pip install wooldridge

     |████████████████████████████████| 5.1 MB 7.3 MB/s 


In [ ]:
import wooldridge as wd

In [ ]:
df = wd.data('wage1')

In [ ]:
wage = df['wage']

In [ ]:
educ = df['educ']

In [ ]:
wage

0       3.10
1       3.24
2       3.00
3       6.00
4       5.30
       ...  
521    15.00
522     2.27
523     4.67
524    11.56
525     3.50
Name: wage, Length: 526, dtype: float64

In [ ]:
educ

0      11
1      12
2      11
3       8
4      12
       ..
521    16
522    10
523    15
524    16
525    14
Name: educ, Length: 526, dtype: int64

In [ ]:
est = Estadisticos(wage.to_list())

In [ ]:
est.media()

5.896102674787035

In [ ]:
est.mediana()

4.650000095367432


In [ ]:
est.varianza()

13.638884363775

In [ ]:
est.desviacion_estandar()

3.6930860217134125

In [ ]:
est.coeficiente_variacion()

0.6263605343078266

In [ ]:
est2 = Estadisticos(educ.to_list())

In [ ]:
est2.media()

12.562737642585551

In [ ]:
est2.mediana()

12.0


In [ ]:
est2.varianza()

7.667485062466015

In [ ]:
est2.desviacion_estandar()

2.769022401943692

In [ ]:
est2.coeficiente_variacion()

0.22041552412566315

##Wage
La base de datos muestra que las personas reciben un sueldo promedio de 5.9$.
Tambien muestra que la media esta por encima de la mediana lo que significa una asimetria posititva en los datos de los sueldos. La desviacion estandar nos dice que algunos datos se alejan de la media en 3.69, confirmandose asi en la varianza que los datos se encuentran proximos a 1 ya que arroja un resultado de 0.62 mostrando una media dispersa.
##Education
La educacion tiene una media de 12.56 estando tambien por encima de la mediana ya que esta es de 12 mostrando unas asimetria positiva. La desviacion y varianza son menores a la de los sueldo demostrando asi que los datos estan mas concentrados en su media, por lo tanto el coeficiente de variacion esta mas proximo a 0 

